src/devices/SenseHat/SenseHatLedMatrix.cs

```diff
@@ -95,9 +144,300 @@ public static int PositionToIndex(int x, int y)
+        public double TextScrollPixelsPerSecond
+        {
+            get
+            {
+                return _textScrollPixelsPerSecond;
+            }
+            set
+            {
+                _textScrollPixelsPerSecond = value;
+                StartOrStopTextScrolling();
+            }
+        }
+
+        /// <summary>
+        /// Text rotation, counterclockwise.
+        /// </summary>
+        [Property]
+        public SenseHatTextRotation TextRotation
+        {
+            get
+            {
+                return _textRotation;
+            }
+            set
+            {
+                _textRotation = value;
+
+                // Apply new state. Lock because render state is nullable and assignment may not be atomic.
+                SenseHatTextRenderState? renderState;
+                lock (_lockTextRenderState)
+                {
+                    if (_textRenderState != null)
+                    {
+                        _textRenderState = _textRenderState.ApplyTextRotation(value);
+                    }
+
+                    renderState = _textRenderState;
+                }
+
+                RenderText(renderState);
+            }
+        }
+
+        private void StartOrStopTextScrolling()
+        {
+            SenseHatTextRenderState? renderState;
+            // Lock because render state is nullable and assignment may not be atomic
+            lock (_lockTextRenderState)
+            {
+                renderState = _textRenderState;
+            }
+
+            int millisecondsPerPixel;
+            if (renderState == null || renderState.TextRenderMatrix.Text.Length == 0 || _textScrollPixelsPerSecond <= 0)
+            {
+                millisecondsPerPixel = 0;
+            }
+            else
+            {
+                millisecondsPerPixel = (int)(1000 / _textScrollPixelsPerSecond);
+            }
+
+            if (millisecondsPerPixel <= 0)
+            {
+                StopTextAnimationTimer();
+            }
+            else
+            {
+                // Calculate the number of milliseconds for one pixel shift
+                StartTextAnimationTimer((int)millisecondsPerPixel);
+            }
+        }
+
+        private void StartTextAnimationTimer(int intervalMs)
+        {
+            StopTextAnimationTimer();
+            _textAnimationTimer = new System.Timers.Timer(intervalMs);
+            _textAnimationTimer.Elapsed += TextAnimationTimer_Elapsed;
+            _textAnimationTimer.Start();
+        }
+
+        private void StopTextAnimationTimer()
+        {
+            if (_textAnimationTimer != null)
+            {
+                _textAnimationTimer.Stop();
+                _textAnimationTimer.Elapsed -= TextAnimationTimer_Elapsed;
+                _textAnimationTimer.Dispose();
+                _textAnimationTimer = null;
+            }
+        }
+
+        private void TextAnimationTimer_Elapsed(object? sender, System.Timers.ElapsedEventArgs e)
```

Comment:
Rather than timer consider allowing user pick when drawing happens. I.e.:

```
TextScroller scroller = ....;

while (true)
{
  scroller.Draw(); // note that you should keep track of time separately inside of this method so it doesn't move inconsistently when user calls this at not equal intervals
  Thread.Sleep(20);
}
```

this kind of design is usually more

Reply:
And note - if you make this design the way I described above you can still use Timer in your app but it won't be necessary for scroller to have dependency on it - you'd just call DrawFrameIfNecessary inside of the timer callback

Reply:
(of course pick better names if you have any idea but ideally it should be straight forward to understand what the method is doing by just looking at the name)

Reply:
Maybe this could be an extended feature in the future, where the user can toggle between automatic scrolling and manual scrolling? For SenseHat, there is probably much gain in pausing text because then only one character (or parts of two) are visible at a time, but it would be a good thing to have for longer displays.